lib/puppet/defaults.rb

```diff
@@ -604,14 +604,28 @@ def self.default_vendormoduledir
       :type       => :duration,
       :desc       => "The minimum time to wait between checking for updates in
       configuration files.  This timeout determines how quickly Puppet checks whether
-      a file (such as manifests or templates) has changed on disk. #{AS_DURATION}",
+      a file (such as manifests or puppet.conf) has changed on disk. The default will
+      change in a future release to be 'unlimited', requiring a reload of the Puppet
+      service to pick up changes to its internal configuration. To reparse files
+      within an environment in Puppet Server please use the environment_cache endpoint",
+      :hook => proc do |val|
+        if ![0, '15s'].include?(val)
+          Puppet.deprecation_warning(<<-WARNING)
```

Comment:
This should be a squiggly heredoc `<<~WARNING` so that the leftmost whitespace is trimmed